### DISCUSSION TITLE
hardhat-etherscan ignore/skip comments

### DISCUSSION THREAD

**Discussion by derawi** [NomicFoundation/hardhat#3440] [Ideas]
Is is somehow possible to skip some comments from being uploaded to etherscan, which are for internal use only?

**Comment by fvictorio** [comment]
Well, that depends a lot on what you are doing. But one idea that comes to mind is using [`hardhat-preprocessor`](https://github.com/wighawag/hardhat-preprocessor) to eliminate the comments. For example, if you want to erase all the comments that start with `// temp:` you could install and add the plugin, and put something like this in your config:

```js
  preprocess: {
    eachLine: hre => ({
      transform: line => {
        return line.includes("// temp:") ? line.replace(/\s*\/\/ temp:.*/, "") : line
      }
    })
  }
```

To be honest, I would avoid doing this, because it could cause some unexpected issues down the line. As a contrived example: if you add a string value that happens to have `// temp:` inside it, you will get a compilation error (because the string will be erased from the middle). That's a very unlikely thing to happen, but the point is that you _could_ get compilation errors or, worse, unexpected runtime behaviors that are super hard to understand because you are preprocessing the code.

That plugin also has a `preprocess` task that will modify your contracts in-place. I think this is a bit better: you could do that only before verifying your contracts, manually inspect that everything looks good, verify your contracts and then undo those changes.

I hope that helps!
